### ISSUE TITLE
Re-opened dialog doesn't fire EventTypeDialogResponse

### ISSUE THREAD

**Issue by kchmie** [kodeyeen/omp#2]
Hi! I have the following code:
```go
func PromptLogin(p *omp.Player, hash string) {
	dialog := omp.NewPasswordDialog("Witaj!", _p("Znaleźliśmy twoje konto w bazie danych. \nWpisz poniżej swoje hasło aby się zalogować."), "Zaloguj", _p("Wyjdź"))
	dialog.Listen(omp.EventTypeDialogResponse, func(e *omp.InputDialogResponseEvent) bool {
		if e.Response == omp.DialogResponseRight {
			p.Kick()
			return false
		}

		err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(e.InputText))

		if err == nil {
			p.SendClientMessage("zalogowano", 0xFFFFFFFF)
		} else {
			p.SendClientMessage(_p("złe hasło"), 0xFFFFFFFF)
			PromptLogin(p, hash)

			return false
		}

		return true
	})
	dialog.ShowFor(p)
}
```
In the first iteration of the function, everything goes as expected. Howewer, when the `PromptLogin` function is fired from within the response event, the displayed dialog will not dispatch an event anymore. It also happens without the bcrypt function.

What I expect is for it to re-show the dialog when the password is incorrect and handle further iterations. 
What I get is the event being fired only the first time. The second time, the dialog just disappears without dispatching an event.

I've looked through the library's code but can't notice anything that would cause this. You know the code the best so I just opened an issue.

**Comment by hexxfy** [comment]
Fixed. Thanks for reporting.

By the way you can optimize your code a little bit.
You're creating a password dialog and registering an event handler for it every time player enters a wrong password.
Instead of `PromptLogin(p, hash)` you could just call again `dialog.ShowFor(p)`.

It would be nice to be able to access that dialog in the event object itself via `e.Dialog`.
I'll probably add this feature later.

**Comment by kchmie** [comment]
Fixed as of [v0.3.1](https://github.com/kodeyeen/omp/releases/tag/v0.3.1), closing. 😀
